src/main/java/pl/edu/mimuw/DepositBankAction.java

```diff
@@ -1,9 +1,62 @@
 package pl.edu.mimuw;
 
-public abstract class DepositBankAction extends BankAction {
+import java.sql.Timestamp;
+import java.util.concurrent.TimeUnit;
 
-  // TODO:
-  // - remove abstract modifier from this class
-  // - this class has to contain extra fields other than LoanBankAction
-  // - you have to design it's constructors/accessors/setters and implement some from BankAction
+public class DepositBankAction extends BankAction {
+    private final double interestRate;
+    private final double capitalizationRate;
+    private boolean collected = false;
+
+    public DepositBankAction(double initial, Timestamp start,
+                             Timestamp end, String id, double interestRate,
+                             double capitalizationRate) {
+        super(initial, start, end, id);
+        this.interestRate = interestRate;
+        this.capitalizationRate = capitalizationRate;
+    }
+
+    public boolean isEligible(BankClient client) {
+        return client.getBalance() >= initialAmount;
+    }
+
+    public void execute(Bank bank, BankClient client) {
```

Comment:
to bardzo ryzykowne co napisałeś - z jednaj strony bank zarzadza klientami i oni akcjami, a z drugiej akcje klientami? Tak to jest zbyt powiązany można by powiedzieć kod (błędne koło trochę) - spróbuj to poprawić jeżeli będzie okaazja